service/raft/group0_state_machine.cc

```diff
@@ -306,26 +309,71 @@ future<> group0_state_machine::transfer_snapshot(raft::server_id from_id, raft::
     std::optional<service::raft_snapshot> raft_snp;
 
     if (_topology_change_enabled) {
-        auto auth_tables = db::system_keyspace::auth_tables();
-        std::vector<table_id> tables;
-        tables.reserve(3);
-        tables.push_back(db::system_keyspace::topology()->id());
-        tables.push_back(db::system_keyspace::topology_requests()->id());
-        tables.push_back(db::system_keyspace::cdc_generations_v3()->id());
+        auto split_into_topology_and_raft_snapshot = [] (service::raft_snapshot&& snp) -> std::pair<service::raft_snapshot, service::raft_snapshot> {
+            auto it = std::partition(snp.mutations.begin(), snp.mutations.end(), [] (const canonical_mutation& m) {
+                    return m.column_family_id() == db::system_keyspace::topology()->id()
+                            || m.column_family_id() == db::system_keyspace::topology_requests()->id()
+                            || m.column_family_id() == db::system_keyspace::cdc_generations_v3()->id();
+            });
 
-        topology_snp = co_await ser::storage_service_rpc_verbs::send_raft_pull_snapshot(
-            &_mm._messaging, addr, as, from_id, service::raft_snapshot_pull_params{std::move(tables)});
+            service::raft_snapshot topology_snp;
+            service::raft_snapshot raft_snp;
 
-        tables = std::vector<table_id>();
-        tables.reserve(auth_tables.size());
+            std::move(snp.mutations.begin(), it, std::back_inserter(topology_snp.mutations));
+            std::move(it, snp.mutations.end(), std::back_inserter(raft_snp.mutations));
 
-        for (const auto& schema : auth_tables) {
-            tables.push_back(schema->id());
-        }
-        tables.push_back(db::system_keyspace::service_levels_v2()->id());
+            return {std::move(topology_snp), std::move(raft_snp)};
+        };
+
+        if (_snapshot_rpc_receiver_side_enabled) {
+            slogger.trace("transfer snapshot: requesting everything at once using the new RPC semantics");
```

Comment:
I'd do this on INFO level. This is not a frequent operation. But it's a big one.
(the `trace` at the beginning of this function could also be INFO.)

Reply:
Done, I increased verbosity of this log message, the first one in the "else" branch corresponding to this if, and also of the first "snapshot transfer" log message in this function.